src/journal/bsod.c

```diff
@@ -280,7 +311,9 @@ static int run(int argc, char *argv[]) {
                 return 0;
         }
 
-        r = display_emergency_message_fullscreen((const char*) message);
+       message_id = TAKE_PTR(id);
```

Comment:
This causes memory leak. Please directly pass `id` to `display_emergency_message_fullscreen()`.